src/toolkit/res_manip.cc

```diff
@@ -65,6 +68,34 @@ std::vector<Resource::Ptr> ResCast(std::vector<Resource::Ptr> rs) {
   return rs;
 }
 
+double GetFillMass(Resource::Ptr r, Package::Ptr pkg) {
+  if (r->quantity() == 0) {
+    throw Error("cannot fill zero resources into a package");
+  }
+  else if (r->quantity() < pkg->fill_min()) {
+    // less than one pkg of material available
+    return 0;
+  }
+
+  std::vector<Resource::Ptr> rs;
+  Resource::Ptr r_pkgd;
+  double fill_mass;
+  if (pkg->strategy() == "first") {
+    fill_mass = pkg->fill_max();
+  } else if (pkg->strategy() == "equal") {
+    int num_min_fill = std::floor(r->quantity() / pkg->fill_min());
+    int num_max_fill = std::ceil(r->quantity() / pkg->fill_max());
+    if (num_min_fill >= num_max_fill) {
```

Comment:
This is really slick and compact logic, but I confess that it's not obvious to me when this will NOT be true? Maybe we should put an example in a comment somewhere?

Reply:
Yeah, it occurs when there's leftover material that cannot be packaged even keeping every package at `fill_min`

Example: 
`quantity` = 5
`fill_min` = 3
`fill_max` = 4

`num_min_fill` = floor(5/3) = 1
`num_max_fill` = ceil(5/4) = 2

Basically, there's not enough to fill two packages at min fill mas, but you have more than one max filled package worth of material. The solution is you fill one package, and then have to return the remaining (in this case, 1) to the buffer-- it cannot be packaged given the packaging strategy